src/utils.rs

```diff
@@ -1,168 +1,217 @@
-//! Utility date and time equations for Temporal
+#![allow(
+    unused,
+    reason = "prefer to have unused methods instead of having to gate everything behind features"
+)]
 
-use alloc::format;
-use alloc::string::String;
+//! Utility date and time equations for Temporal
+// NOTE: Potentially add more tests.
 
 use crate::MS_PER_DAY;
+pub(crate) const MS_PER_HOUR: i64 = 3_600_000;
+pub(crate) const MS_PER_MINUTE: i64 = 60_000;
 
 mod neri_schneider;
 
-pub(crate) use neri_schneider::epoch_days_from_gregorian_date;
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct Epoch {
```

Comment:
thought: Potentially rename to `EpochMilliseconds` and upgrade to `pub`.

We have an `EpochNanoseconds` type that's already public, and I was already debating whether to have a public `EpochMillisecond` for `Instant::from_milliseconds` and use `EpochNanoseconds` in `Instant::from_nanoseconds`. This could then result in pulling a majority of the `EpochMilliseconds` type out of `utils` into something like `epoch_primitives`, `epoch`, or something. I'm not really sure about the module name. I'm not in love with where `EpochNanoseconds` is exported from currently either, which I think is `temporal_rs::time`. Heck, `temporal_rs::epoch` probably makes even more sense than `time`.

Doing this then leaves whatever is left in `utils` as pure utility functions.